tests/foreman/cli/test_katello_agent.py

```diff
@@ -46,288 +42,291 @@
-        ActivationKey.add_host_collection(
-            {
-                'id': activation_key['id'],
-                'organization-id': self.org['id'],
-                'host-collection-id': hc['id'],
-            }
-        )
-        # add the registered instance host to collection
-        HostCollection.add_host(
-            {'id': hc['id'], 'organization-id': self.org['id'], 'host-ids': self.host['id']}
-        )
-        with VirtualMachine() as client:
-            client.create()
-            client.install_katello_ca()
-            # register the client host with the current activation key
-            client.register_contenthost(self.org['name'], activation_key=activation_key['name'])
-            assert client.subscribed
-            # note: when registering the host, it should be automatically added
-            # to the host collection
-            client_host = Host.info({'name': client.hostname})
-            hosts = HostCollection.hosts({'id': hc['id'], 'organization-id': self.org['id']})
-            assert len(hosts) == 2
-            expected_hosts_ids = {self.host['id'], client_host['id']}
-            hosts_ids = {host['id'] for host in hosts}
-            assert hosts_ids == expected_hosts_ids
+from robottelo.hosts import ContentHost
+
+
+@skip_if_not_set('clients', 'fake_manifest')
+@pytest.fixture(scope='module')
+def katello_agent_repos(module_ak, module_cv, module_lce, module_org):
+    """Create Org, Lifecycle Environment, Content View, Activation key"""
+    setup_org_for_a_rh_repo(
+        {
+            'product': PRDS['rhel'],
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
+            'organization-id': module_org.id,
+            'content-view-id': module_cv.id,
+            'lifecycle-environment-id': module_lce.id,
+            'activationkey-id': module_ak.id,
+        }
+    )
+    # Create custom repository content
+    setup_org_for_a_custom_repo(
+        {
+            'url': FAKE_1_YUM_REPO,
+            'organization-id': module_org.id,
+            'content-view-id': module_cv.id,
+            'lifecycle-environment-id': module_lce.id,
+            'activationkey-id': module_ak.id,
+        }
+    )
+    return {
+        'ak': module_ak,
+        'cv': module_cv,
+        'lce': module_lce,
+        'org': module_org,
+    }
+
+
+@skip_if_not_set('clients')
+@pytest.fixture
+def katello_agent_client(katello_agent_repos, rhel7_contenthost):
+    rhel7_contenthost.install_katello_ca()
+    # Register content host and install katello-agent
+    rhel7_contenthost.register_contenthost(
+        katello_agent_repos['org'].label,
+        katello_agent_repos['ak'].name,
+    )
+    assert rhel7_contenthost.subscribed
+    host_info = Host.info({'name': rhel7_contenthost.hostname})
+    rhel7_contenthost.enable_repo(REPOS['rhst7']['id'])
+    rhel7_contenthost.install_katello_agent()
+    yield {'client': rhel7_contenthost, 'host_info': host_info}
```

Comment:
Same here, this fixture can just return `host_info`, and tests could reference `rhel7_contenthost` as provided through its own fixture.